### PR TITLE
Feature: virtual decorator

### PR DESCRIPTION
**Pull request by LotfiMEZIANI**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #1349

## What is the new behavior?

Described here : #1349

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by kamilmysliwiec** [comment]
Can you add tests?

**Comment by LotfiMEZIANI** [comment]
> Can you add tests?

It's done. But i don't know if I do it in the correct folder (`tests/e2e`).

**Comment by hobbydevs** [comment]
Would love to see this get merged.

File: lib/storages/type-metadata.storage.ts [hunk 1]
```diff
@@ -33,6 +39,10 @@ export class TypeMetadataStorageHost {
   ) {
     return this.properties.filter(belongsToClass);
   }
+
+  getVirtualsMetadataByTarget<TClass>(targetFilter: Type<TClass>) {
```
Comment (TbotaPhantA) [inline lib/storages/type-metadata.storage.ts:43]: Isn't it better to add `virtuals` to `SchemaMetadata` and then create method `getClassVirtualsByPredicate` like the one below ?

**Comment by bytesbay** [comment]
Any updates here?

**Review by bytesbay — APPROVED** [review]
clear enough

**Comment by LotfiMEZIANI** [comment]
@kamilmysliwiec, waiting for the merge, if everything is ok.

**Comment by KevinEdry** [comment]
@LotfiMEZIANI  @kamilmysliwiec Any news on this? this should have merged by now...

**Comment by LotfiMEZIANI** [comment]
@KevinEdry, for me it’s fine, I’m waiting for the validation of an owner of the repository (ping @kamilmysliwiec )

**Comment by graddaniel** [comment]
bump

I just happen to need this.

**Comment by souravdasslg** [comment]
@kamilmysliwiec Can you please look into this ?

**Comment by Youcef-Z** [comment]
@kamilmysliwiec could you merge this please 🙏

**Comment by LhonRafaat** [comment]
After 2 years xd

**Comment by kamilmysliwiec** [comment]
Better late than never, I guess

**Pull request merged**